cmd/gateway-s3.go

```diff
@@ -344,15 +344,14 @@ func (l *s3Objects) GetObjectInfo(bucket string, object string) (objInfo ObjectI
 
 // PutObject creates a new object with the incoming data,
 func (l *s3Objects) PutObject(bucket string, object string, data *HashReader, metadata map[string]string) (objInfo ObjectInfo, err error) {
-	sha256sumBytes, err := hex.DecodeString(data.sha256Sum)
+	sha256sumBytes, err := hex.DecodeString(data.SHA256Sum)
```

Comment:
I believe SHA256 and MD5 hashes are required to be string with fixed encoding.  You could have respective methods in `HashReader` than public fields.

Reply:
This change be done..